packages/hardhat-core/test/internal/solidity/compilation-job.ts

```diff
@@ -6,7 +6,10 @@ import {
 } from "../../../src/internal/solidity/compilation-job";
 import { ResolvedFile } from "../../../src/internal/solidity/resolver";
 import * as taskTypes from "../../../src/types/builtin-tasks";
-import { CompilationJobCreationError } from "../../../src/types/builtin-tasks";
+import {
+  CompilationJobCreationError,
+  CompilationJobCreationErrorReason,
+} from "../../../src/types/builtin-tasks";
```

Comment:
This file needs more tests, but I will add them after the new types are definitive.